src/main/groovy/io/seqera/wave/service/blob/impl/BlobCacheServiceImpl.groovy

```diff
@@ -228,6 +263,25 @@ class BlobCacheServiceImpl implements BlobCacheService {
         return result ? Waiter.awaitCompletion(blobStore, key, result) : null
     }
 
+    /**
+     * Await for the container layer blob download
+     *
+     * @param key
+     *      The container blob unique key
+     * @param uri
+     *      uri pointing to blob0
+     *
+     * @return
+     *      the {@link java.util.concurrent.CompletableFuture} holding the {@link BlobCacheInfo} associated with
+     *      specified blob key or {@code null} if no blob record is associated for the
+     *      given key
+     */
+    BlobCacheInfo awaitCacheStore(String key, String uri) {
+        final result = blobStore.getBlob(key)
+        result.locationUri = uri
+        return result ? Waiter.awaitCompletion(blobStore, key, result) : null
+    }
```

Comment:
Is it needed to keep the version `BlobCacheInfo awaitCacheStore(String key)` ? think can be replaced by this latter

Reply:
i have removed it

Reply:
Improved it a bit